src/query/atomWithQuery.ts

```diff
@@ -11,28 +11,62 @@ import type { Getter, PrimitiveAtom, WritableAtom } from 'jotai'
 import { queryClientAtom } from './queryClientAtom'
 
 export type AtomWithQueryAction = { type: 'refetch' }
-
+export type CreateQueryOptions<Options> = Options | ((get: Getter) => Options)
+export type GetQueryClient = (get: Getter) => QueryClient
 export type AtomWithQueryOptions<TQueryFnData, TError, TData, TQueryData> =
```

Comment:
Types seem to work correctly now, so this is ok as is.
I just expected to omit `'enabled'` from this `AtomWithQueryOptions` type like in L27, which seems safer? what do you think?

Reply:
I played around with that and a few other options, but this was the result that allowed for all the previous atoms in the tests to not change their typings, but correctly change the return value of the new test atoms from a guaranteed return value to one that is potentially `undefined`. 

My understanding of how this is currently set up is that this type provides the option for `enabled` to be passed, while `AtomWithQueryOptionsWithEnabled` then is used when `enabled` is passed (for any value, true/false). When I remove it from AtomWithQueryOptions, it causes type errors on L144 and another, thinking `enabled` is never a correct option to pass.

Reply:
Thanks. I didn't dig into it, and if you did it's good to go.